packages/-ember-data/tests/integration/adapter/find-test.js

```diff
@@ -144,32 +160,52 @@ module('integration/adapter/find - Finding Records', function(hooks) {
       })
     );
 
-    let store = this.owner.lookup('service:store');
+    const store = this.owner.lookup('service:store');
 
-    return run(() => {
-      return store.findRecord('person', 1).catch(reason => {
-        assert.ok(true, 'The rejection handler was called');
-        assert.ok(!store.hasRecordForId('person', 1), 'The record has been unloaded');
-      });
-    });
+    try {
+      await store.findRecord('person', '1');
+      assert.ok(false, 'We expected to throw but did not');
+    } catch (e) {
+      assert.ok(true, 'The rejection handler was called');
+      assert.ok(!store.hasRecordForId('person', 1), 'The record has been unloaded');
+    }
   });
 
-  testInDebug('When a single record is requested, and the payload is blank', function(assert) {
+  testInDebug('When a single record is requested, and the payload is blank', async function(assert) {
+    const Person = Model.extend({
+      name: attr('string'),
+    });
+
+    this.owner.register('model:person', Person);
+    this.owner.register('adapter:application', Adapter.extend());
+    this.owner.register('serializer:application', JSONAPISerializer.extend());
     this.owner.register(
       'adapter:person',
       Adapter.extend({
         findRecord: () => resolve({}),
       })
     );
 
-    let store = this.owner.lookup('service:store');
+    const store = this.owner.lookup('service:store');
 
-    assert.expectAssertion(() => {
-      run(() => store.findRecord('person', 'the-id'));
-    }, /You made a 'findRecord' request for a 'person' with id 'the-id', but the adapter's response did not have any data/);
+    try {
+      await store.findRecord('person', 'the-id');
+      assert.ok(false, 'We expected to throw but did not');
+    } catch (e) {
+      const expectedMessageRegex = /You made a 'findRecord' request for a 'person' with id 'the-id', but the adapter's response did not have any data/;
```

Comment:
this particular error feels like we could do a strictEquals comparison and be cleaner